File: Perl/whale.pl
#!/usr/bin/perl -w

$whale_name = $ARGV[0];
$nb_of_pod = 0;
$nb_of_indiv = 0;

@lines = <STDIN>;

foreach $var (@lines){
	if($var =~ m/[0-9]+ $whale_name/){
		$nb_of_pod++;
		$var =~ s/ [a-z]+.*$//;
		$nb_of_indiv += $var;
	}
}

print "$whale_name observations: $nb_of_pod pods, $nb_of_indiv individuals\n";

File: Python/whale.py
#!/usr/bin/python3

import sys
import re

target_whale = sys.argv[1]

record = sys.stdin.readlines()

nb_of_pods = 0
nb_of_individuals = 0

for r in record:
    if target_whale in r:
        nb_of_pods += 1
        nb = re.sub(r'[^\d]*', '', r)
        nb_of_individuals += int(nb)



print('{} observations: {} pods, {} individuals'.format(target_whale,
                                                        nb_of_pods,
                                                        nb_of_individuals))
    
